### ISSUE TITLE
Core dump due to workaround for Qt bug

### ISSUE THREAD

**Issue by pmattern** [tsujan/FeatherPad#10]
FeatherPad reproducibly crashes and dumps core right when checkbox "Workaround for Qt5's scroll jumb bug" in dialogue "Preferences" is ticked. When it had been launched from a terminal emulator there's an error message
> ASSERT failure in QList<T>::at: "index out of range", file /usr/include/qt/QtCore/qlist.h, line 537

Traces can be found in [this Gist](https://gist.github.com/pmattern/866f6ada2d0c570e9822a924072dbe62).

Seen running 7b0502b on Arch Linux.

Btw. there's probably a typo in the said option's text which should say "jum*p*".

**Comment by tsujan** [comment]
Thank you!
Although that option is always checked here (I hate those scroll jumps), I can't reproduce the issue. But will surely look into it!

**Comment by tsujan** [comment]
I fixed a very bad typo that *may* have been the cause (https://github.com/tsujan/FeatherPad/commit/08eb2fe8d9672c8533d68ffb15f8cc42f2e7305e).

There should be something strange in my system that prevents such crashes. It can't be about software... Hardware? I have no clue.

**Comment by pmattern** [comment]
Problem can no longer be seen as of 08eb2fe. Also, it cannot be seen with FeatherNotes at all which makes it even more likely that it was caused by that typos.

**Comment by tsujan** [comment]
Yes, those typos should have caused it. The mystery remains why the crash didn't happen here -- my system just ignored the problem!

**Comment by pmattern** [comment]
> The mystery remains why the crash didn't happen here

This is indeed weird.
I assume you're not talking about some missing traces as displayed by `coredumpctl list` or `coredumpctl gdb <PID>` but the application itself really didn't crash and kept running at your side, right?
One way to reproduce the crashes 100% here was start FeatherPad upon deleting `~/.config/featherpad/`, tick checkbox "Workaround for Qt5's scroll jumb bug", file some text and save as file, repeat last two steps in another tab, uncheck and check the said checkbox.
Did you have debugging enabled while building (`CONFIG+=debug`)? Just a guess, not too likely this matters.

**Comment by tsujan** [comment]
> ... deleting ~/.config/featherpad/, tick checkbox "Workaround for Qt5's scroll jumb bug", file some text and save as file, repeat last two steps in another tab, uncheck and check the said checkbox.

I hadn't tested that scenario (which causes a crash with the typo in question -- tested just now) but several other ones that, theoretically, should lead to a similar crash. Even if I was tired when testing and so made mistakes (which is improbable), I'm sure that https://github.com/tsujan/FeatherPad/issues/6 wasn't reproducible here. A few minutes ago, I reversed https://github.com/tsujan/FeatherPad/commit/9f667ba533492d02a92a9795a64c9f7cb8b7c28b just to test again and yet no crash! Actually, I also tested memory usage several times and became sure that extra FeatherPad windows were not deleted after being closed in my case but they were in yours (with the same Qt version, I suppose).

As for debugging, yes, I enable it but, as you said, it shouldn't matter.
